FogCreek/Glitch-Community
# 🎏: punycode-ify project, user, team names

## Links
* Remix link: https://🎏.glitch.me/
* Manuscript link: : https://glitch.manuscript.com/f/cases/3328347/unescaped-characters-in-user-team-names

## Changes
* escape project, user, team names with punycode in URL and API calls
* escape collection name with `encodeURIComponent` in server-side API calls
* continue to render names in their escaped form (eg. 🎏 as `xn--jk8h`) outside of the URL because this is essentially an easter egg; we just want it to stop raising errors 

## How To Test
- open the console on https://glitch.com/edit/#!/xn--jk8h (the project is owned by `@community`, you can add yourself)
- verify the following pages render, and don't raise any errors in the console:
  * home: https://🎏.glitch.me
  * project: https://🎏.glitch.me/~🎏
  * team: https://🎏.glitch.me/@🚞
  * collections: https://🎏.glitch.me/@🚞/🚡
